src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -329,8 +372,109 @@ class HuiMapCard extends LitElement implements LovelaceCard {
       }
     }
 
+    // DRAW history
+    if (config.history_entities && this._stateHistory?.timeline) {
+      const colors = [
```

Comment:
These colors should probably not be hardcoded. So that users can theme them. But if they do need to be in an array they should be in `src\data\map.ts` Which you will have to create

Reply:
We have a list of colors here, we should use them.
<https://github.com/home-assistant/frontend/blob/dev/src/components/entity/ha-chart-base.js#L540>

Reply:
> 
> 
> These colors should probably not be hardcoded. So that users can theme them. But if they do need to be in an array they should be in `src\data\map.ts` Which you will have to create

Is there a special way to provide theme colors, or just via the configuration file?

Reply:
> 
> 
> We have a list of colors here, we should use them.
> https://github.com/home-assistant/frontend/blob/dev/src/components/entity/ha-chart-base.js#L540

Nice. Maybe it is useful to move them to a separate file, to provided central color information?